include/nccl_ofi.h

```diff
@@ -37,10 +38,10 @@ extern "C" {
 
 /*
  * We have a limit of MAX_HANDLE_SIZE = 64 bytes. Therefore, we can only
- * support an endpoint name of maximum 56 bytes. We are using remaining
- * 8 bytes for tags.
+ * support an endpoint name of maximum 32 bytes. We are using remaining
+ * 32 bytes for tags and saving communicator stage.
  */
-#define MAX_EP_ADDR		(56)
+#define MAX_EP_ADDR		(32)
```

Comment:
What happens if we try to mix old and new NCCL when two ranks try to connect?  I'm assuming we are not going to get a sane error message.

I wonder if we can add versioning if we keep this unchanged but add additional information to the connect message (see below comments for more details).

Reply:
Do you mean 2 different NCCL version ranks trying to connect with each other? If `MAX_EP_ADDR` is small then we will fail when using `fi_getname` for any provider so it should never reach the stage of actual communication.

I thought about adding versioning but to be honest we were overestimating the needs for `MAX_EP_ADDR` in current plugin versions. I haven't come across a provider which would need to provide more than 32 bytes for now.

Reply:
I am worried about eventually needing more information that will fit into 64 bytes, so I'd prefer to have versioning in the handle.  Can we spare 1 byte?

Also, if we know MAX_EP_ADDR is overestimated today, does that mean we can effectively guarantee that some bytes are 0?  Maybe we can make one of these the version byte (and the non-versioned current variant magically becomes "version 0").

Reply:
If we had zero-ed out the buffer then yes some bytes were 0. I think we are very tight on handle size here so having a versioning in the handle is costly.

Wouldn't it be easier to just change the size of the handle in NCCL if we foresee that we need more room for `EP_ADDR`?

Reply:
Doesn't that directly contradict your comment above?  :) I feel like only one of these statements can be true.

I won't push too hard for this, but I think getting NVIDIA to change the handle size may be a challenge.  Also, a handle of a different size doesn't solve the versioning problem for plugin versions prior to the size change.

Reply:
I am also thinking out loud to see if it would be better to just change the NCCL handle size to a larger size to accommodate future changes to libfabric providers (No intention to contradict myself :)).

If the handle size is big enough, then why would you need a versioning of handle?

Reply:
In case it becomes not big enough in the future :)

Reply:
Opened a PR for increasing handle size: https://github.com/NVIDIA/nccl/pull/635